### DISCUSSION TITLE
Preparing for v5.1

### DISCUSSION THREAD

**Discussion by axelboc** [axelboc/anki-ultimate-geography#563] [General]
I'm finally on holidays 🎉 and I have a couple of days on my hands with not much planned, so I thought I'd get v5.1 out! It's been 11 months since v5.0, so there's a lot of stuff in the [v5.1 milestone](https://github.com/anki-geo/ultimate-geography/milestone/9?closed=1). It's been a busy year for me, but things are starting to settle down again, so hopefully the next one won't be as far away.

I don't recall any breaking changes or anything that may require special upgrade instructions. That being said, it may be good to review the wiki's [Upgrade instructions](https://github.com/anki-geo/ultimate-geography/wiki/Upgrade-instructions) and maybe mention v5.1 somewhere (maybe tell people to upgrade to v5 first, or be more generic about versions, like `v5.x` or `v5.0 or newer`)

Does anyone wish to get something through before the release?

**Comment by aplaice** [comment]
> I'm finally on holidays tada

I hope you're enjoying your holidays!

> That being said, it may be good to review the wiki's [Upgrade instructions](https://github.com/anki-geo/ultimate-geography/wiki/Upgrade-instructions) and maybe mention v5.1 somewhere (maybe tell people to upgrade to v5 first, or be more generic about versions, like v5.x

Thanks for updating the instructions in the Wiki! (Changing to v5.x is indeed, I think, the correct approach.)

> Does anyone wish to get something through before the release?

There are many things that I'd wish to get through before the release, but it's unlikely I will, so I don't think it's worth waiting for them. (I don't think we have any critical issues that _have_ to be fixed.)

**Comment by axelboc** [comment]
Draft release notes ready! https://github.com/anki-geo/ultimate-geography/releases/tag/untagged-d8879117fc534616c581 Let me know if you spot any mistake / feel free to edit if you have the rights.

**Reply by aplaice** [reply]
I edited the README link to point to https://github.com/anki-geo/ultimate-geography/blob/master/README.md#upgrading (since AFAIR on mobile "repo" README links (like https://github.com/anki-geo/ultimate-geography#upgrading) don't work).

Unfortunately, the link to the draft was automatically changed:

https://github.com/anki-geo/ultimate-geography/releases/tag/untagged-58b2d5982155d7e869dc

(I think the notes look great! Thanks, as always! :))

**Reply by axelboc** [reply]
I've added the ZIP files. Let me know if it all looks good and I'll publish the release tomorrow.

Note that `ug-flag-afghanistan.png` was renamed to `ug-flag-afghanistan.svg` but the Afghanistan note and `sources.csv` were still using the PNG extension. I noticed it using Anki's "Check media" feature. I've pushed a fix to `master`.

**Reply by aplaice** [reply]
> Note that ug-flag-afghanistan.png was renamed to ug-flag-afghanistan.svg but the Afghanistan note and sources.csv were still using the PNG extension. I noticed it using Anki's "Check media" feature. I've pushed a fix to master.

Oops, thanks! (I'm an idiot — I had updated the source link, but not the file extension, in either place...)

<hr/>

tldr; IMO dealing with this is more trouble than it's worth.

One slight, but annoying, issue is that upgrading from 5.0 to 5.1 invokes CrowdAnki's "Change Note Type" dialog.

Clicking OK results in a correct upgrade, without issues.

However, the dialog is slightly scary and if a user changes any options, they can mess up their notes.

In general, CrowdAnki [invokes](https://github.com/Stvad/CrowdAnki/blob/master/crowd_anki/representation/note_model.py#L74) the dialog when the JSON fragments corresponding to the note-types [differ](https://github.com/Stvad/CrowdAnki/blob/master/crowd_anki/utils/utils.py#L41),

In this case (5.0 to 5.1), the changes are:

1. A typo in the CSS was fixed (rectagular -> rectangular)
2. BrainBrew now inserts the `latexsvg`, `bfont`, `bsize` and `scratchPad` keys (with default values) into the note_models part of the JSON (this is sensible behaviour, since these keys have been added to Anki a while ago).

We could, in principle:
1. undo the typo fix and 
2. modify BrainBrew to allow not inserting these keys (with some option...) or alternatively modify CrowdAnki to ignore `bfont` et al. if they have the default values.

However, IMO this is overkill.

It would be nice if CrowdAnki reported how exactly the note models differ, but that definitely won't happen before AUG's 5.1 release (if ever)... (It's nice that at least it allows updating the note type/model at all — now testing Anki's built-in import of an APKG with a modified CSS style it apparently doesn't apply the modifications at all and doesn't even warn the user that it's ignoring the changes (???).)

**Reply by aplaice** [reply]
I added a section to the Wiki:


> # From v5.0 to v5.1

> Please follow the standard [normal installation/upgrade procedures](https://github.com/anki-geo/ultimate-geography/blob/master/README.md#upgrading). If you see a "Change Note Type" dialog please click OK without changing any of the mappings (Change "Country - Capital" to "Country - Capital" etc.).

The lower version to v5.x upgrade instructions are now technically out of date for 5.1, so I added a section describing the 5.0->5.1 issue elsewhere where it wouldn't result in overload and just confuse people.

**Reply by axelboc** [reply]
Ah nice, yeah I did wonder if it was just me, and then forgot about it. I wrote something in the release notes as well.

**Reply by axelboc** [reply]
Should we revert the 5.x references to 5.0?

**Reply by aplaice** [reply]
Given that the horrible GUID mapping and the use of the legacy deck to separate out cards will still apply just as well for ->5.1 not just ->5.0 and that there's no practical reason to upgrade stepwise (u.v -> 5.0, 5.0->5.1) other than making it easier for us to describe things, I don't think so (but feel free to do so if you think it'd make things less confusing!).
